### PR TITLE
Fix two issues with MetaflowCode and get_data

### PR DESCRIPTION
**Pull request by msavela**
- `get_data` properly handling generator iteration
- `MetaflowCode` accidentally tried to init `BytesIO` with tuple instead of data blob

### CODE REVIEW

**Comment by romain-intel** [comment]
As mentioned offline: there is another similar issue l112 on conda_environment.py. We can address that one here and merge. Changes LGTM.

**Comment by msavela** [comment]
Good find, pushed additional fix at `conda_environment.py`

File: metaflow/plugins/conda/conda_environment.py [hunk 1]
```diff
@@ -109,8 +109,9 @@ def get_client_info(cls, flow_name, metadata):
         if info is None or env_id is None:
             return {'type': 'conda'}
         info = json.loads(info)
-        with cls._filecache.get_data(info['ds_type'], flow_name, info['location'], info['sha']) as f:
-            with tarfile.open(fileobj=f, mode='r:gz') as tar:
+        with cls._filecache.get_data(info['ds_type'], flow_name, info['location'], info['sha']) as data:
```
Comment (romain-intel) [inline metaflow/plugins/conda/conda_environment.py:112]: I actually don't think we need the with here, it's just _, blob = cls._filecache and then blob = BytesIO(blob) (I think)

Comment (msavela) [reply metaflow/plugins/conda/conda_environment.py]: Made changes based on your suggestion

**Review by romain-intel — APPROVED** [review]

**Pull request merged**